patchwork-registries/src/main/java/net/minecraftforge/registries/ForgeRegistry.java

```diff
@@ -39,23 +39,98 @@
 import net.minecraft.util.Identifier;
 import net.minecraft.util.registry.DefaultedRegistry;
 import net.minecraft.util.registry.Registry;
+import net.minecraft.util.registry.MutableRegistry;
 
-public class ForgeRegistry<V extends IForgeRegistryEntry<V>> implements IForgeRegistry<V> {
+import net.fabricmc.fabric.api.event.registry.RegistryEntryAddedCallback;
+
+import net.patchworkmc.impl.registries.RemovableRegistry;
+import net.patchworkmc.impl.registries.ForgeModDefaultRegistry;
+import net.patchworkmc.impl.registries.ForgeModRegistry;
+import net.patchworkmc.impl.registries.VanillaRegistry;
+
+public class ForgeRegistry<V extends IForgeRegistryEntry<V>> implements
+		IForgeRegistryModifiable<V>, IForgeRegistryInternal<V>, RegistryEntryAddedCallback<V> {
 	public static Marker REGISTRIES = MarkerManager.getMarker("REGISTRIES");
 	private static Logger LOGGER = LogManager.getLogger();
-	private Identifier name;
-	private Registry<V> vanilla;
-	private Class<V> superType;
-
-	public ForgeRegistry(Identifier name, Registry<V> vanilla, Class<V> superType) {
+	private final Identifier name; // The forge name
+	private final boolean isVanilla;
+	private final Registry<V> vanilla;
+	private final Class<V> superType;
+	private final Map<Identifier, ?> slaves = new HashMap<>();
+	private final CreateCallback<V> create;
```

Comment:
all these callbacks could use a `Callback` suffix, so `create` -> `createCallback` etc. It makes the code a bit easier to follow

Reply:
fixed